docs/site/pages/assets/dsl.mdx

```diff
@@ -0,0 +1,126 @@
+
+
+## Creating a Basic Component
+
+The `Asset` component from the `@player-tools/dsl` package is the quickest way to create a new component. The `Asset` component will take all the Asset's properties and convert them to their equivalent JSON representation when serialized.
+
+In the examples below, we'll be creating a TSX component for the `action` asset in our reference set. 
+
+The `action` asset has a `label` slot (which is typically used as a `text` asset), a `value` (for flow transitions), and an `exp` for evaluating expressions.
+It's type resembles something akin to:
+
+```ts
+import { Asset, AssetWrapper, Expression } from '@player-ui/player';
+
+export interface ActionAsset<AnyTextAsset extends Asset = Asset> extends Asset<'action'> {
+  /** The transition value of the action in the state machine */
+  value?: string;
+
+  /** A text-like asset for the action's label */
+  label?: AssetWrapper<AnyTextAsset>;
+
+  /** An optional expression to execute before transitioning */
+  exp?: Expression;
+}
+```
+
+To turn this interface into a usable component, create a new React component that _renders_ an Asset:
+
+```tsx
+import { Asset, AssetPropsWithChildren } from '@player-tools/dsl';
+
+export const Action = (props: AssetPropsWithChildren<ActionAsset>) => {
+  return <Asset type="action" {...props} />;
+}
+```
+
+This would allow users to import the `Action` component, and _render_ it to JSON:
+
+```tsx
+const myView = <Action value="next" />
+```
+
+The `AssetPropsWithChildren` type is a utility type to help convert the `Asset` type (which has a required `id` and `type` properties) to a type more suited for components. It changes the `id` to be optional, and adds a `applicability` property automatically.
+
+## Slots
+
+Continuing the example fo the `ActionAsset`, we need a way for users to users to specify the nested `label` property, which itself is another asset. This can be accomplished using the `createSlot` utility function.
+
+```tsx
+import { Asset, AssetPropsWithChildren } from '@player-tools/dsl';
+
+export const Action = (props: AssetPropsWithChildren<ActionAsset>) => {
+  return <Asset type="action" {...props} />;
+}
+
+Action.Label = createSlot({
+  name: 'label'
+})
+```
+
+This adds component (`Action.Label`) that will automatically place any nested children under the `label` property of the parent asset:
+
+```tsx
+const myView = (
+  <Action value="next">
+    <Action.Label>
+      <Text value="Continue" />
+    </Action.Label>
+  </Action>
+);
+```
+
+It also accepts components for automatically creating `text` and `collection` assets, allowing users to shortcut the `Text` component:
```

Comment:
Meaning if there was no shortcut for the Text/Collection creation? It would compile the string as the value of the Label property which would cause the content to fail at validation.